scipy/linalg/_decomp.py

```diff
@@ -729,7 +730,7 @@ def eig_banded(a_band, lower=False, eigvals_only=False, overwrite_a_band=False,
         multiplicity.
     v : (M, M) float or complex ndarray
         The normalized eigenvector corresponding to the eigenvalue w[i] is
-        the column v[:,i].
+        the column v[:,i]. Only returned if ``eigvals_only=False``.
```

Comment:
You don't have to warn for default behavior. `eigvals_only` is `True` by default.